tests/baselines/reference/contextuallyTypedParametersWithInitializers.errors.txt

```diff
@@ -0,0 +1,48 @@
+tests/cases/compiler/contextuallyTypedParametersWithInitializers.ts(8,27): error TS7022: '{ foo = 42 }' implicitly has type 'any' because it does not have a type annotation and is referenced directly or indirectly in its own initializer.
+tests/cases/compiler/contextuallyTypedParametersWithInitializers.ts(8,29): error TS7031: Binding element 'foo' implicitly has an 'any' type.
+tests/cases/compiler/contextuallyTypedParametersWithInitializers.ts(14,27): error TS7006: Parameter 'foo' implicitly has an 'any' type.
+tests/cases/compiler/contextuallyTypedParametersWithInitializers.ts(27,40): error TS7022: '{ foo = 42 }' implicitly has type 'any' because it does not have a type annotation and is referenced directly or indirectly in its own initializer.
+tests/cases/compiler/contextuallyTypedParametersWithInitializers.ts(27,42): error TS7031: Binding element 'foo' implicitly has an 'any' type.
+
+
+==== tests/cases/compiler/contextuallyTypedParametersWithInitializers.ts (5 errors) ====
+    declare function id1<T>(input: T): T;
+    declare function id2<T extends (x: any) => any>(input: T): T;
+    declare function id3<T extends (x: { foo: any }) => any>(input: T): T;
+    declare function id4<T extends (x: { foo?: number }) => any>(input: T): T;
+    declare function id5<T extends (x?: number) => any>(input: T): T;
+    
+    const f10 = function ({ foo = 42 }) { return foo };
+    const f11 = id1(function ({ foo = 42 }) { return foo });  // Implicit any error
+                              ~~~~~~~~~~~~
+!!! error TS7022: '{ foo = 42 }' implicitly has type 'any' because it does not have a type annotation and is referenced directly or indirectly in its own initializer.

```

Comment:
How does this binding pattern circularly reference itself? o.O

Reply:
Hmm, strange. This error is caused by declaration file emit, it only appears with `--declaration`. Have no idea why.

Reply:
`getEmitDeclarations`  causes us to conditionally `collectLinkedAliases` on exported names - but this is a binding pattern in an anonymous function expression, not an exported name... o.O

Reply:
Maybe we generate the error while looking up the type of the expression for emitting `declare const f11: something`? We marked `foo` as implicitly `any` within the checker, however did we propagate that up to the binding pattern as a whole?

Reply:
The circularity error is caused by the parameter having itself as its contextual type. It's actually unrelated to this PR. For example, this causes the same error on `master`:

```ts
declare function id<T>(input: T): T;
const f1 = id(function ({ foo }) { return foo });
```